redbpf/src/lib.rs

```diff
@@ -2280,7 +2357,8 @@ impl<'a> SockMap<'a> {
         Ok(SockMap { base: map })
     }
 
-    pub fn set(&mut self, mut idx: u32, mut fd: RawFd) -> Result<()> {
+    pub fn set(&mut self, mut idx: u32, fd: RawFd) -> Result<()> {
+        let mut fd = fd as u64;
```

Comment:
@raftario 
I'm not sure why you changed this.
Can you elaborate?

Reply:
That was a temporary fix of mine, I tested my changes with the code from [this talk](https://ebpf.io/summit-2020-slides/eBPF_Summit_2020-Lightning-Jakub_Sitnicki-Steering_connections_to_sockets_with_BPF_socke_lookup_hook.pdf) which uses `sizeof(__u64)` as the value size for its socket map, so passing it a pointer to a 32bit value caused errors. This can probably be removed given the value size should be `sizeof(int)`.